### ISSUE TITLE
Loss Decline very slowly in  wmt_ende_tokens_32k

### ISSUE THREAD

**Issue by sptravel** [tensorflow/tensor2tensor#36]
training on a single nVidia 1080 GTX (12 GB) with the following parameters:
--problems=wmt_ende_tokens_32k 
--model=transformer 
--hparams_set=transformer_big_single_gpu 
--hparams='batch_size=2048'

When the loss value is around 3, the loss value drops very slowly：
step 0~ 30k：loss drops from 8.4 to 3
INFO:tensorflow:global_step/sec: 2.03158
INFO:tensorflow:loss = 3.62252, step = 29601 (49.221 sec)
INFO:tensorflow:global_step/sec: 2.03539
INFO:tensorflow:loss = 3.64336, step = 29701 (49.130 sec)
INFO:tensorflow:global_step/sec: 2.03153
INFO:tensorflow:loss = 3.58582, step = 29801 (49.226 sec)
INFO:tensorflow:global_step/sec: 2.02831
INFO:tensorflow:loss = 3.38816, step = 29901 (49.301 sec)
INFO:tensorflow:global_step/sec: 2.02674
INFO:tensorflow:loss = 3.40213, step = 30001 (49.340 sec)
INFO:tensorflow:global_step/sec: 2.03157
INFO:tensorflow:loss = 3.44571, step = 30101 (49.235 sec)
INFO:tensorflow:global_step/sec: 2.03308
INFO:tensorflow:loss = 3.15277, step = 30201 (49.175 sec)

step 120k：loss  is jitter around 3
INFO:tensorflow:loss = 3.12874, step = 125101 (76.470 sec)
INFO:tensorflow:global_step/sec: 2.04413
INFO:tensorflow:loss = 3.09151, step = 125201 (48.925 sec)
INFO:tensorflow:global_step/sec: 2.03683
INFO:tensorflow:loss = 3.2518, step = 125301 (49.093 sec)
INFO:tensorflow:global_step/sec: 2.03616
INFO:tensorflow:loss = 3.90474, step = 125401 (49.113 sec)
INFO:tensorflow:global_step/sec: 2.04036
INFO:tensorflow:loss = 2.87875, step = 125501 (49.010 sec)
INFO:tensorflow:global_step/sec: 2.0414
INFO:tensorflow:loss = 3.47175, step = 125601 (48.986 sec)
INFO:tensorflow:global_step/sec: 2.03132
INFO:tensorflow:loss = 3.00751, step = 125701 (49.230 sec)
INFO:tensorflow:global_step/sec: 2.0305
INFO:tensorflow:loss = 2.81739, step = 125801 (49.247 sec)
INFO:tensorflow:global_step/sec: 2.03291
INFO:tensorflow:loss = 3.60361, step = 125901 (49.191 sec)
INFO:tensorflow:global_step/sec: 2.03915
INFO:tensorflow:loss = 2.91831, step = 126001 (49.041 sec)
INFO:tensorflow:global_step/sec: 2.02992
INFO:tensorflow:loss = 2.98262, step = 126101 (49.263 sec)
INFO:tensorflow:global_step/sec: 2.03459

Is it normal?
Can you give a reference value?

**Comment by lukaszkaiser** [comment]
Attached is a figure of the loss of transformer_big (not base) on wmt_ende_32k trained on 8 gpus. 8 gpus mean that the batch is 8x larger, so the loss decreases faster. 
![transformer_big_loss](https://user-images.githubusercontent.com/684901/27506008-03e15386-5864-11e7-82c5-b8e586334cf4.png)

Below is the corresponding figure with bpe-tokenized BLEU (not the paper BLEU, beware). But you can see that it gets most of it by 30k steps as well, and that a loss around 1.7 is already good, and around 1.5 is very good.
![transformer_big_bleu](https://user-images.githubusercontent.com/684901/27506023-65edb7f4-5864-11e7-9188-469b6a9819d6.png)

I'll get the figures for 1-gpu base next week. One thing to note: the t2t-trainer is able of running evals, we should probably set it to run an eval every few hundred steps (currently it only does so after local training, you need a second process with --train_steps=0 --eval_steps=100 to start every few minutes). If you run the eval job though, you can start tensorboard and you'll see metrics on the dev set like these:
![transformer_big_metrics](https://user-images.githubusercontent.com/684901/27506035-c1b00010-5864-11e7-9490-369bf69850c2.png)

I hope this helps a bit for now.

**Comment by tobyyouup** [comment]
@lukaszkaiser   Do I need to start the eval process manually every few minutes. Is there some params that we can use to automatically run the eval steps?

**Comment by ggsonic** [comment]
train transformer_big on 4 gpus, using default params with --worker_gpu=4,get similar results. loss is around 3, metrics/bleu is around 0.18. i thought the learning rate might be high so  i also try with learning_rate=0.05,learning_rate_warmup_steps=16000, still loss is around 3. is there some bugs in the transformer related code? can not reproduce the results like paper said.

**Comment by lukaszkaiser** [comment]
How many steps did you run? I just trained 3 separate `transformer_big` models on 8 gpus and I'm getting metrics/bleu from 41.7 to 41.9 at about 180k steps. It's very strange you're getting 18, how many steps was that? What GPU are you using? Let me try to replicate your setup as close as possible.
